File: src/main/java/de/blazemcworld/fireflow/command/JoinCommand.java
package de.blazemcworld.fireflow.command;

import de.blazemcworld.fireflow.space.SpaceInfo;
import de.blazemcworld.fireflow.space.SpaceManager;
import de.blazemcworld.fireflow.util.Transfer;
import de.blazemcworld.fireflow.util.Translations;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;
import net.minestom.server.command.builder.Command;
import net.minestom.server.command.builder.arguments.number.ArgumentInteger;
import net.minestom.server.entity.Player;

public class JoinCommand extends Command {
    
    public JoinCommand() {
        super("join");

        addSyntax((sender, ctx) -> {
            if (!(sender instanceof Player player)) {
                sender.sendMessage(Component.text(Translations.get("error.needs.player")).color(NamedTextColor.RED));
                return;
            }
            SpaceInfo info = SpaceManager.info.get(ctx.get("id"));
            if (info == null) {
                sender.sendMessage(Component.text(Translations.get("error.invalid.space")).color(NamedTextColor.RED));
                return;
            }

            Transfer.move(player, SpaceManager.getOrLoadSpace(info).play);
        }, new ArgumentInteger("id"));
    }

}

File: src/main/kotlin/de/blazemcworld/fireflow/command/JoinCommand.kt
package de.blazemcworld.fireflow.command

import de.blazemcworld.fireflow.database.table.SpacesTable
import de.blazemcworld.fireflow.space.SpaceManager
import de.blazemcworld.fireflow.util.sendError
import net.minestom.server.command.builder.Command
import net.minestom.server.command.builder.arguments.number.ArgumentInteger
import net.minestom.server.entity.Player
import org.jetbrains.exposed.sql.SqlExpressionBuilder.eq
import org.jetbrains.exposed.sql.selectAll
import org.jetbrains.exposed.sql.transactions.transaction

object JoinCommand : Command("join") {
    init {
        addSyntax(exec@{ sender, ctx ->
            if (sender !is Player) return@exec
            val id = ctx.get<Int>("space_id")

            if (SpaceManager.currentSpace(sender)?.id == id) {
                sender.sendError("You are already on this space!")
                return@exec
            }

            var found = false
            transaction {
                found = !SpacesTable.selectAll().where(SpacesTable.id eq id).empty()
            }

            if (!found) {
                sender.sendError("Unknown space id!")
                return@exec
            }

            SpaceManager.sendToSpace(sender, id)
        }, ArgumentInteger("space_id"))
    }
}